### PR TITLE
[ECO-2536] Refactor the emoji picker to not skip flags and new emojis, make it more robust and easy to use

### PR DESCRIPTION
**Pull request by xbtmatt**
# Description

- [x] Remove the check for flags. The check for emoji support is nifty, but since we now have a font that will support all emojis we use, we don't need it (noto-color-emoji). This is why we couldn't see lots of flags in the picker
- [x] Remove the check for newer versions. This hid a lot of emojis from versions 13/14/15 from the user. They will now be visible 
- [x] Ideally we'd keep the check in for support, but because we're using a custom font, it's too much of a lift to use a custom font in a canvas context (which is how they verify in the picker library, once per version; aka if one emoji from version 15 isn't supported, all from version 15 are considered unsupported and thus would not show in the picker)
- [x] Add scripts to build the package `yarnpack:all`
- [x] Separate filtering emojis initially from later (emojis you don't want to show up in the picker at all, e.g. chat emojis while registering or searching for a market)
- [x] Add prop for disabling an emoji with a callback. This lets us conditionally disable emojis if the input would put a market symbol over the max symbol length
- [x] Add native styles that we were updating in the shadow DOM. This is generally significantly more performant and cleaner than what we were doing
- [x] Clean everything up, add as little code as possible to prevent a big rebase if the version updates and we need to bump it.
- [x] Remove all formatting changes
- [x] Remove `pnpm` and continue to use `yarn`

# Testing

Loooots of manual testing.

### CODE REVIEW

File: packages/emoji-mart/src/components/Picker/PickerStyles.scss [hunk 1]
```diff
@@ -54,18 +54,18 @@
...
   width: 100%;
   text-align: left;
   border-radius: var(--border-radius);
   background-color: rgb(var(--em-rgb-background));
+  
+  @media (prefers-reduced-motion) {
+    --duration: 0;
+    --duration-fast: 0;
+    --duration-instant: 0;
+  }
```
Comment (xbtmatt) [inline packages/emoji-mart/src/components/Picker/PickerStyles.scss]: Moved to satisfy new SASS/scss warnings when compiling with pnpm. Feel free to resolve once this is seen- only noting so it's not confusing why this seemingly meaningless change is here.

File: packages/emoji-mart/src/index.ts [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
-export { FrequentlyUsed, SafeFlags, SearchIndex, Store } from './helpers'
+export { FrequentlyUsed, SafeFlags, SearchIndex, Store, NativeSupport } from "./helpers";
 
-export { init, Data, I18n } from './config'
+export { init, Data, I18n } from "./config";
 
-export { getEmojiDataFromNative } from './utils'
+export { getEmojiDataFromNative } from "./utils";
+
+console.log(`This package was built at ${new Date().toLocaleString()}`);
```
Comment (xbtmatt) [inline packages/emoji-mart/src/index.ts]:
```suggestion
```

**Comment by CRBl69** [comment]
It would be nice to have the least amount of changes as possible compared to the original repo in case we ever need to rebase.

I would undo the formatting changes.

Also, add new logic in separate files to simplify rebasing event more.

File: packages/emoji-mart/src/helpers/search-index.ts [hunk 3]
```diff
@@ -1,83 +1,89 @@
...
-      if (score == -1) continue
-
-      results.push(emoji)
-      scores[emoji.id] || (scores[emoji.id] = 0)
-      scores[emoji.id] += emoji.id == value ? 0 : score + 1
+      if (!alreadyChecked.has(emoji.id)) {
+        alreadyChecked.add(emoji.id);
+        if (!emoji.search) {
+          console.debug(`Emoji ${emoji.id} doesn't have a \`search\` field.`);
+        }
```
Comment (xbtmatt) [inline packages/emoji-mart/src/helpers/search-index.ts]:
```suggestion
```

**Review by CRBl69 — APPROVED** [review]
LGTM :ok_hand:.

Just make sure this works with our repo by pointing the frontend to your branch before merging.

**Pull request merged**